lib/puppet/provider/mongodb_replset/mongo.rb

```diff
@@ -275,6 +275,7 @@ def self.mongo_command(command, host=nil, retries=4)
 
     #Hack to avoid non-json empty sets
     output = "{}" if output == "null\n"
+    output = "{}" if output == "\nnull\n"
```

Comment:
If a mongo connection with --sslAllowInvalidHostnames is made, it prints a warning like:
`2017-05-16T10:56:27.679+0200 warning: The server certificate does not match the host name 127.0.0.1`
This warning gets filtered but there is still a newline and if the return is null, the content will be:
`"\nnull\n"`
So this just extends the already implemented logic to still work with the new feature.